libr/arch/arch_session.c

```diff
@@ -0,0 +1,152 @@
+/* radare - LGPL - Copyright 2020 - pancake */
+
+#include <r_arch.h>
+#include <r_asm.h>
+#include "../config.h"
+
+
+static bool r_arch_session_set_cpu(RArchSession *ai, const char *cpu) {
+	r_return_val_if_fail (ai && cpu, false);
+	if (ai && *cpu) {
+		// TODO: improve the check with r_str_split_list()
+		if (R_STR_ISEMPTY (ai->cur->cpus) || strstr (ai->cur->cpus, cpu)) {
+			free (ai->setup.cpu);
+			ai->setup.cpu = cpu? strdup (cpu): NULL;
+		}
+		return true;
+	}
+	return false;
+}
+
+static bool r_arch_session_set_syntax(RArchSession *ai, RArchSyntax syntax) {
+	r_return_val_if_fail (ai, false);
+	switch (syntax) {
+	case R_ASM_SYNTAX_REGNUM:
```

Comment:
shouldn't we have `R_ARCH_SYNTAX_*`?

Reply:
yes, as well as many other constants from r_util, r_asm and r_anal, that should be rebranded underr the r_arch umbrella. but imho we shuold do this change later, otherwise this PR can grow too much